Backpack/src/androidTest/java/net/skyscanner/backpack/button/BpkButtonTest.kt

```diff
@@ -24,23 +35,39 @@ class BpkButtonTest {
   // The drawables are set as start,top,end,bottom and are accessible in the compoundDrawables array
   @Test
   fun test_icon_end() {
-    val context = InstrumentationRegistry.getInstrumentation().targetContext
     val trainIcon = AppCompatResources.getDrawable(context, R.drawable.bpk_train)
     val button = BpkButton(context).apply {
       icon = trainIcon
       iconPosition = BpkButton.END
     }
-    Assert.assertEquals(button.compoundDrawables[2],trainIcon)
+    Assert.assertEquals(button.compoundDrawables[2], trainIcon)
   }
 
   @Test
   fun test_icon_start() {
-    val context = InstrumentationRegistry.getInstrumentation().targetContext
     val trainIcon = AppCompatResources.getDrawable(context, R.drawable.bpk_train)
     val button = BpkButton(context).apply {
       icon = trainIcon
       iconPosition = BpkButton.START
     }
-    Assert.assertEquals(button.compoundDrawables[0],trainIcon)
+    Assert.assertEquals(button.compoundDrawables[0], trainIcon)
+  }
+
+  @Test
+  fun givenBpkButtonInAnyTypeAndFormWhenIsEnabledIsSetThenSetupIsCalled() {
+    // Given
+    val subjectUnderTest = BpkButton(context).apply {
+      isEnabled = true
+      type = BpkButton.Type.Primary
+    }
+    val newState = false
+    val disabledBackgroundColor = R.color.bpkGray100
+
+    // When
+    subjectUnderTest.isEnabled = newState
+
+    // Then
+    assert(subjectUnderTest.isEnabled == newState)
```

Comment:
Does this give a readable error? I suspect `Assert.assertEquals` will give an error with more information